File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/lighthouses/domain/LighthousesObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.lighthouses.domain;


public abstract class LighthousesObject {
    public abstract String objAsString();
    public static LighthousesObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new LighthousesMarkerObject();
            case "lighthouse":
                return new LighthousesLighthouseObject();
            default:
                return new LighthousesMarkerObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/lighthouses/domain/LighthousesObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.lighthouses.domain

abstract class LighthousesObject {
    abstract fun objAsString(): String

    companion object {
        fun objFromString(str: String?): LighthousesObject {
            return when (str) {
                "marker" -> LighthousesMarkerObject()
                "lighthouse" -> LighthousesLighthouseObject()
                else -> LighthousesMarkerObject()
            }
        }
    }
}